Source/Private/ASMutableElementMap.mm

```diff
@@ -108,6 +102,34 @@ - (void)insertElement:(ASCollectionElement *)element atIndexPath:(NSIndexPath *)
   }
 }
 
+- (void)migrateSupplementaryElementsWithChangeSet:(_ASHierarchyChangeSet *)changeSet
+{
+  if (changeSet.deletedSections.count == 0 && changeSet.insertedSections.count == 0) {
+    return;
+  }
+
+  // For each element kind,
+  [_supplementaryElements enumerateKeysAndObjectsUsingBlock:^(NSString * _Nonnull key, NSMutableDictionary<NSIndexPath *,ASCollectionElement *> * _Nonnull supps, BOOL * _Nonnull stop) {
+    
+    // For each index path of that kind,
+    [[supps copy] enumerateKeysAndObjectsUsingBlock:^(NSIndexPath * _Nonnull oldIndexPath, ASCollectionElement * _Nonnull obj, BOOL * _Nonnull stop) {
+      NSInteger oldSection = oldIndexPath.section;
+      NSInteger newSection = [changeSet newSectionForOldSection:oldSection];
+
+      if (oldSection == newSection) {
+        // Index path stayed the same, do nothing.
+      } else if (newSection == NSNotFound) {
+        // Section was deleted, remove the supplementary element.
+        [supps removeObjectForKey:oldIndexPath];
+      } else {
+        // Section index changed, move it.
+        NSIndexPath *newIndexPath = [NSIndexPath indexPathForItem:oldIndexPath.item inSection:newSection];
```

Comment:
I think in this case we may also need to do the removeObjectForKey:oldIndexPath ?